packages/lesswrong/server/scripts/migrateCollections.ts

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-console */
+import type { ITask } from "pg-promise";
+import { Vulcan, getCollection } from "../vulcan-lib";
+import { forEachDocumentBatchInCollection } from "../manualMigrations/migrationUtils";
+import { getSqlClientOrThrow } from "../../lib/sql/sqlClient";
+import Table from "../../lib/sql/Table";
+import CreateTableQuery from "../../lib/sql/CreateTableQuery";
+import CreateIndexQuery from "../../lib/sql/CreateIndexQuery";
+import InsertQuery from "../../lib/sql/InsertQuery";
+import SwitchingCollection from "../../lib/SwitchingCollection";
+
+type Transaction = ITask<{}>;
+
+// Custom formatters to fix data integrity issues on a per-collection basis
+// A place for nasty hacks to live...
+const formatters: Partial<Record<CollectionNameString, (document: DbObject) => DbObject>> = {
+  Posts: (post: DbPost): DbPost => {
+    const scoreThresholds = [2, 30, 45, 75, 125, 200];
+    for (const threshold of scoreThresholds) {
+      const prop = `scoreExceeded${threshold}Date`;
```

Comment:
Why is this necessary? Are these not normal fields?

Reply:
This is just cleaning up garbage values in that exist in the database due to old bugs and that violate the Postgres schema. As we migrate more collections, we'll probably find more data integrity issues like this (and Jim thinks LessWrong will have even more due to the size and age of their database).